### DISCUSSION TITLE
Create multiple feature branches but only merge last feature branch with develop

### DISCUSSION THREAD

**Discussion by gitbaby2018** [community/community#84157] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello all, 

Let's say I created a feature branch 'A' from develop, did some work but did not merge with develop. Then I created a branch 'B' off of branch 'A' , did some work but also did not merge with develop. Now let's say I created branch 'C' off of 'B', did some work and only merged 'C' with develop. Will now C contain all changes of branches B and A? If yes, after the merge it will be safe to delete branches A, B, and C?

Is the below workflow correct in my scenario?

git checkout develop
git checkout -b A
git commit -am "all A changes"
git push

git checkout -b B A
git commit -am "all B changes"
git push

git checkout -b C B
git commit -am "all C changes"
git push

git checkout develop
git merge -m "merging C with develop" C

**Comment by bonizario** [comment]
Yes, your understanding is correct. If you create branch 'C' from 'B', and 'B' was created from 'A', then 'C' will contain all the changes from 'A' and 'B'. When you merge 'C' into 'develop', all changes from 'A', 'B', and 'C' will be included in 'develop'.

After the merge, it is safe to delete branches 'A', 'B', and 'C' if you no longer need them. You can delete a branch with `git branch -d <branch-name>`. If the branch has not been merged and you still want to delete it, you can use `git branch -D <branch-name>`.

Your workflow is correct for this scenario. Just make sure to push your changes to the remote repository if you're working in a team or on multiple machines. You can do this with `git push origin <branch-name>` after each commit. 🚀

**Reply by gitbaby2018** [reply, marked as answer]
thank you @bonizario
